### PR TITLE
prov/rxm: Mark the funcs as static if it isn't used outside the file

### PR DESCRIPTION
**Pull request by gladkovdmitry17**
Also this PR includes:
- Replace rxm_[get|set]opt by fi_no_[get|set]opt from fi_enosys

Signed-off-by: Dmitry Gladkov <dmitry.gladkov@intel.com>

### CODE REVIEW

File: prov/rxm/src/rxm_ep.c [hunk 1]
```diff
@@ -833,16 +821,16 @@ rxm_ep_send_common(struct fid_ep *ep_fid, const struct iovec *iov, void **desc,
...
 }
 
 static ssize_t rxm_ep_send(struct fid_ep *ep_fid, const void *buf, size_t len,
 			   void *desc, fi_addr_t dest_addr, void *context)
 {
-	struct iovec iov;
-	iov.iov_base = (void *) buf;
-	iov.iov_len = len;
+	struct iovec iov = {
+		.iov_base = (void *)buf,
```
Comment (shefty) [inline prov/rxm/src/rxm_ep.c]: If you declare iov as 'const struct iovec', can we remove the buf cast to (void*) that is used to toss away the const from the buf definition?  This comment applies everywhere this cast is done.

Comment (gladkovdmitry17) [reply prov/rxm/src/rxm_ep.c]: I've checked, this produces warning anyway, for instance:
```
prov/rxm/src/rxm_ep.c:831:3: warning: initialization discards 'const' qualifier from pointer target type [enabled by default]
   .iov_base = buf,
```

Comment (shefty) [reply prov/rxm/src/rxm_ep.c]: Thanks for checking.

**Comment by gladkovdmitry17** [comment]
@a-ilango Seems, the PR is ready for review. Please, take a look into these minor changes in the RxM.

**Comment by a-ilango** [comment]
The changes look good but Travis CI is failing linux gcc build. The issue looks like a false positive though.

**Comment by gladkovdmitry17** [comment]
> The changes look good but Travis CI is failing linux gcc build. The issue looks like a false positive though.

@a-ilango @shefty Thanks! I re-pushed the commits with no updates. Let's wait when CI testing is completed.

**Pull request merged**